Bundle each task and link in pipelines

Michkov opened the pull request:
@jduimovich - This is prototype of pushing each task into separate bundle. Deployed under my namespace -> https://quay.io/user/mkovarik .

It's prototype, only devfile pipeline is working. Pipelines bundle - quay.io/mkovarik/build-templates-bundle:2022-01-18-101924 -> all tasks are pointing to bundle based on task name and tag, which is same as for build-templates-bundle (in this case 2022-01-18-101924).

Tested with `./hack/build/build-deploy.sh  https://github.com/devfile-samples/devfile-sample-code-with-quarkus`.

For building own version it's enough to call `./dev.sh`. Only problem is that when repository is firstly created in quay it's private for some reason, so I had to public it before running `build.sh`. Also it makes quey org slightly messy, so there could be option to deploy it into separate org just with the tasks.

Michkov committed:
Bundle each task and link in pipelines

* Push tasks into tags of one image
* Update pipelines to use tasks in bundles
* Modify test-all-tasks.sh to use kustomize

jduimovich commented:
I think the 1  pipeline per bundle and 1 task per bundle will be messy and have too many containers.
I hope that the bundles in Tekton can be evolved to have more than 10 tasks/pipelines.
We could also consider a single org for tasks and use the tag name as the task name + a version
eg
appstudio-tasks:taskname-v0.1.3  which then means you don't get the weird quay/public/private issue everytime you create a new task, we only do that once  for the `appstudio-tasks` org. When bundles expand, we can move all into a single bundle and remove the taskname from the tag

Michkov committed:
Add script for releasing

One script which can release appstudio-utils, appstudio-tasks and
build-templates-bundle with the same tag. The tag has to be the same tag
as git tag of current branch.

Michkov committed:
Update README.md with new distribution way
